index.bs

```diff
@@ -522,6 +522,158 @@ See also:
+follow the "Discard non-"[=Document/fully active=]" documents for situations that can't be supported" pattern described below.
+  
+Additionally, when a document becomes "[=Document/fully active=]" again,
+it can be useful to update it with the current state of the world,
+if anything has changed while it is in the non-"[=Document/fully active=]" state.
+However, care needs to be taken with events that occurred while in the BFCache.
+When not "[=Document/fully active=]", for some cases, all events should be dropped,
+in some the latest state should be delivered in a single event,
+in others it may be appropriate to queue events or deliver a combined event.
+The correct approach is case by case and should consider privacy,
+correctness, performance and ergonomics.
+
+<div class="example">
+  The [gamepadconnected](https://w3c.github.io/gamepad/#event-gamepadconnected) event
+  can be sent to a document that becomes "[=Document/fully active=]" again
+  if a gamepad is connected while the document is not "[=Document/fully active=]".
+  If the gamepad was repeatedly connected and disconnected,
+  only the final connected event should be delivered.
+</div>
+<div class="example">
+  For geolocation or other physical sensors,
+  no information about what happened while not-"[=Document/fully active=]" should be delivered.
+  The events should simply resume from when the document became "[=Document/fully active=]"
+</div>
+<div class="example">
+  For network connections or streams,
+  the data received while not-"[=Document/fully active=]" should be delivered only
+  when the document becomes "[=Document/fully active=]" again,
+  but whereas a stream might have created many events with a small amount of data each,
+  it could be delivered as smaller number of events with more data in each.
+</div>
+
+<h4 id="omit-non-fully-active">Omit non-"[=Document/fully active=]" documents from APIs that span multiple documents</h4>
+Non-"[=Document/fully active=]" documents should not be observable,
+so APIs should treat them as if they no longer exist.
+They should not be visible to the "outside world" through document-spanning APIs
+(e.g. [clients.matchAll()](https://w3c.github.io/ServiceWorker/#clients-matchall), window.opener).
+
+Note: This should be rare since cross-document-spanning APIs are themselves relatively rare.
+
+<div class="example">
+  [BroadcastChannel](https://html.spec.whatwg.org/multipage/web-messaging.html#broadcasting-to-other-browsing-contexts:fully-active)
+  checks for "[=Document/fully active=]" before sending messages to other browsing contexts.
+</div>
+<div class="example">
+  clients.matchAll()'s [spec](https://w3c.github.io/ServiceWorker/#clients-matchall)
+  currently does not distinguish between "[=Document/fully active=]"
+  and non-"[=Document/fully active=]" clients,
+  but correct implementations should only return "[=Document/fully active=]" clients.
```

Comment:
Can we link to a spec issue about getting this fixed?

Reply:
Linked to https://github.com/w3c/ServiceWorker/issues/1594 now